core/src/main/java/org/wildfly/plugin/core/ContainerDescription.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.plugin.core;
+
+/**
+ * Information about the running container.
+ *
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+@SuppressWarnings({"WeakerAccess", "unused"})
+public interface ContainerDescription {
+
+    /**
+     * Returns the name of the product.
+     *
+     * @return the name of the product
+     */
+    String getProductName();
+
+    /**
+     * Returns the product version, if defined, or {@code null} if the product version was not defined.
+     *
+     * @return the product version or {@code null} if not defined
+     */
+    String getProductVersion();
+
+    /**
+     * Returns the release version, if defined, or {@code null} if the release version was not defined.
+     * <p>
+     * Note that in WildFly 9+ this is usually the version for WildFly Core. In WildFly 8 this is the full version.
+     * </p>
+     *
+     * @return the release version or {@code null} if not defined
+     */
+    String getReleaseVersion();
+
+    /**
+     * Returns the type of the server that was launched.
+     *
+     * @return the type of the server that was launched or {@code null} if not defined
+     */
+    String getLaunchType();
```

Comment:
It may not be consistent depending on the target container. For example WildFly 8 doesn't have `SELF_CONTAINED`. This API isn't meant to be a full management API. Just a helper for some small things.